dokka-integration-tests/gradle/build.gradle.kts

```diff
@@ -1,41 +1,237 @@
 /*
  * Copyright 2014-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
-
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+import org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+import org.gradle.language.base.plugins.LifecycleBasePlugin.VERIFICATION_GROUP
+import org.jetbrains.kotlin.gradle.dsl.ExplicitApiMode.Disabled
 
 plugins {
-    id("dokkabuild.test-integration")
+    id("dokkabuild.kotlin-jvm")
+    `test-suite-base`
+    `java-test-fixtures`
+    id("dokkabuild.testing.android-setup")
 }
 
 dependencies {
-    implementation(projects.utilities)
+    api(projects.utilities)
 
-    implementation(kotlin("test-junit5"))
-    implementation(libs.junit.jupiterApi)
-    implementation(libs.junit.jupiterParams)
+    api(libs.jsoup)
 
-    implementation(gradleTestKit())
+    api(libs.kotlin.test)
+    api(libs.junit.jupiterApi)
+    api(libs.junit.jupiterParams)
 
-    implementation(libs.jsoup)
+    api(gradleTestKit())
 }
 
-val aggregatingProject = gradle.includedBuild("dokka")
+kotlin {
+    explicitApi = Disabled
 
-tasks.integrationTest {
-    // pass the property to a test fork
-    project.findProperty("org.jetbrains.dokka.experimental.tryK2")
-        ?.let { systemProperty("org.jetbrains.dokka.experimental.tryK2", it) }
+    compilerOptions {
+        allWarningsAsErrors = false
+        optIn.add("kotlin.io.path.ExperimentalPathApi")
+    }
+}
+
+val aggregatingProject = gradle.includedBuild("dokka")
 
+tasks.integrationTestPreparation {
     dependsOn(aggregatingProject.task(":publishToMavenLocal"))
+}
+
+tasks.withType<Test>().configureEach {
+    dependsOn(tasks.integrationTestPreparation)
 
     environment("DOKKA_VERSION", project.version)
 
-    inputs.dir(file("projects"))
+    maxHeapSize = "2G"
+
+    val useK2 = dokkaBuild.integrationTestUseK2.get()
+
+    useJUnitPlatform {
+        if (useK2) excludeTags("onlyDescriptors", "onlyDescriptorsMPP")
+    }
+
+    systemProperty("org.jetbrains.dokka.experimental.tryK2", useK2)
+
+    setForkEvery(1)
+    dokkaBuild.integrationTestParallelism.orNull?.let {
+        maxParallelForks = it
+    }
+
+    environment("isExhaustive", dokkaBuild.integrationTestExhaustive)
+
+    // allow inspecting projects in temporary dirs after a test fails
+    systemProperty(
+        "junit.jupiter.tempdir.cleanup.mode.default",
+        dokkaBuild.isCI.map { isCi -> if (isCi) "ALWAYS" else "ON_SUCCESS" }.get(),
+    )
+
+    testLogging {
+        exceptionFormat = FULL
+        events(SKIPPED, FAILED)
+        showExceptions = true
+        showCauses = true
+        showStackTraces = true
+    }
```

Comment:
Copied from the integration-test convention plugin - see https://github.com/Kotlin/dokka/pull/3427/files#r1432756930 for reasoning